shared/src/main/scala/com/raquo/domtypes/generic/defs/eventProps/MouseEventProps.scala

```diff
@@ -4,38 +4,41 @@ import com.raquo.domtypes.generic.builders.EventPropBuilder
 
 /**
   * Mouse Events: triggered by a mouse, or similar user actions.
+  * @tparam DomHtmlElementTargetEvent
+  *            An event that has an HTMLElement as `target`.
+  *            This event type has no corresponding type in JS DOM. See our own `ElementTargetEvent` trait.
   */
-trait MouseEventProps[EP[_ <: DomEvent], DomEvent, DomMouseEvent <: DomEvent, DomDragEvent <: DomMouseEvent] { this: EventPropBuilder[EP, DomEvent] =>
+trait MouseEventProps[EP[_ <: DomEvent], DomEvent, DomMouseEvent <: DomEvent, DomDragEvent <: DomMouseEvent, DomHtmlElementTargetEvent <: DomEvent] { this: EventPropBuilder[EP, DomEvent] =>
 
   /**
     * The click event is raised when the user clicks on an element. The click
     * event will occur after the mousedown and mouseup events.
     *
     * MDN
     */
-  lazy val onClick: EP[DomMouseEvent] = eventProp("click")
+  lazy val onClick: EP[DomMouseEvent with DomHtmlElementTargetEvent] = eventProp("click")
```

Comment:
I suspect we can indeed do this, but is there any formal indication that such events will only happen on elements and not e.g. text nodes or SVG elements?

Reply:
Good catch, i think you are right! [MDN](https://developer.mozilla.org/en-US/docs/Web/Events/click) actually only says the target type is `Element`.

Then, I was probably wrong to make `onBlur` and `onFocus` in `FormEventProps` be an `HTMLElementTargetEvent`, as they are also only [defined for target `Element`](https://developer.mozilla.org/en-US/docs/Web/Events/blur). Right?

Reply:
Yeah I think we should start out being somewhat conservative in this kind of thing, I'd be happy with `dom.Element` where MDN says so. If we start relying on "conveniently incorrect" types the eventual withdrawal will be painful... I'm still hoping that the other approach I mentioned in a comment on this issue in outwatch will work better for this.

Reply:
Sounds right, I will update those event props to having target of type `dom.Element`. We can always revisit if there is new information.